src/reaper_osara.cpp

```diff
@@ -4810,6 +4811,62 @@ void cmdInsertMarker(Command* command) {
 	outputMessage(format(translate("marker {} inserted"), number));
 }
 
+void cmdInsertProjectMarker(Command* command) {
+	cmdhInsertProjectMarker(command->gaccel.accel.cmd);
+}
+
+void cmdhInsertTakeMarker(int command) {
+	if(CountSelectedMediaItems(0)>0) {
+		vector<int> preMarkers(CountSelectedMediaItems(nullptr));
+		for(int i = 0; i < preMarkers.size(); ++ i) {
+			MediaItem* item = GetSelectedMediaItem(nullptr, i);
+			MediaItem_Take* take = GetActiveTake(item);
+			preMarkers[i] = GetNumTakeMarkers(take);
+		}
+		Main_OnCommand(command, 0);
+		int addedMarkers = 0;
+		for(int i = 0; i < preMarkers.size(); ++ i) {
+			MediaItem* item = GetSelectedMediaItem(nullptr, i);
+			MediaItem_Take* take = GetActiveTake(item);
+			if (preMarkers[i] < GetNumTakeMarkers(take)) {
+				++ addedMarkers;
+			}
+		}
+		if (addedMarkers == 0) {
+			return; // not inserted
+		} else {
+			// Translators: Reported when using REAPER's quick add take marker action. If more than one take marker is inserted, [] will be replaced with the number of markers. E.G. "2 take markers inserted".
+			outputMessage(format(
+				translate_plural("take marker inserted", "{} take markers inserted", addedMarkers), addedMarkers));
+			return;
+		}
+	}
+}
+
+void cmdInsertTakeMarker(Command* command) {
+	cmdhInsertTakeMarker(command->gaccel.accel.cmd);
+}
+
+void cmdInsertProjectOrTakeMarker(Command* command) {
+	if  (fakeFocus == FOCUS_ITEM && CountSelectedMediaItems(nullptr) > 0)
+		cmdhInsertTakeMarker(42390); // Item: Quick add take marker at play position or edit cursor
+	else
+		cmdhInsertProjectMarker(40157); // Markers: Insert marker at current position
+}
+
+void cmdInsertOrEditMarker(Command* command) {
+	double start, end;
+	GetSet_LoopTimeRange(false, true, &start, &end, false);
+	if (start != end && fakeFocus == FOCUS_ITEM) {
+		Main_OnCommand(43181, 0); // Item: Add/edit take marker at time selection
```

Comment:
The action is called "OSARA: Add project or take marker at cursor (depending on focus)". It doesn't mention time selection. Should it?

Reply:
Urgh, just realised that I renamed the wrong action. Time selection is only relevant when using the add/edit one. Another commit coming up shortly.